scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,434 @@
+		return nil, fmt.Errorf("reference values unavailable")
+	}
+
+	err := json.Unmarshal(refValEndorsement.Attributes, &rv)
+	if err != nil {
+		return nil, err
+	}
+
+	return &rv, nil
+}
+
+// evidenceToComidTriple converts claim set to CoMID value triple
+func evidenceToComidTriple(ec *proto.EvidenceContext) (*comid.ValueTriple, error) {
+	evCorimJson, err := json.Marshal(ec.Evidence.AsMap())
+	if err != nil {
+		return nil, err
+	}
+
+	evComid, err := comidFromJson(evCorimJson)
+	if err != nil {
+		return nil, err
+	}
+
+	return &evComid.Triples.ReferenceValues.Values[0], nil
+}
+
+// compareMeasurements checks if two given comid.Measurement variables are the same.
+func compareMeasurements(refM comid.Measurement, evM comid.Measurement) bool {
+	// RawValue comparison
+	if refM.Val.RawValue != nil {
+		if evM.Val.RawValue == nil {
+			return false
+		}
+
+		refDigest, _ := refM.Val.RawValue.GetBytes()
+		return evM.Val.RawValue.CompareAgainstReference(refDigest, nil)
+	}
+
+	// Digests comparison
+	if refM.Val.Digests != nil {
+		if evM.Val.Digests == nil {
+			return false
+		}
+
+		return evM.Val.Digests.CompareAgainstReference(*refM.Val.Digests)
+	}
+
+	// SVN comparison
+	if refM.Val.SVN != nil {
+		if evM.Val.SVN == nil {
+			log.Debugf("evidence doesn't have SVN")
+			return false
+		}
+
+		if c, ok := evM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+				return c.CompareAgainstRefSVN(*r)
+			} else if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.CompareAgainstRefMinSVN(*r)
+			} else {
+				log.Debugf("unknown refVal SVN type")
+				return false
+			}
+		} else if c, ok := evM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.Equal(*r)
+			}
+			log.Debugf("can't compare TaggedMinSVN against TaggedSVN")
+			return false
+		} else {
+			log.Debugf("unknown evidence SVN type")
+			return false
+		}
+	}
+
+	return true
+}
+
+// AppraiseEvidence confirms if the claims in the evidence match with the provisioned
+// reference values.
+//
+// Appraisal can confirm if the evidence is genuinely generated by AMD
+// hardware and if SEV-SNP enables memory encryption. As such, set the
+// "Hardware" and "RuntimeOpaque" values in the trustworthiness vector;
+// we can't infer other aspects of the vector from SEV-SNP evidence alone.
+func (o EvidenceHandler) AppraiseEvidence(
+	ec *proto.EvidenceContext,
+	endorsementsStrings []string,
+) (*ear.AttestationResult, error) {
+	var (
+		err         error
+		evidenceMap map[string]interface{}
+	)
+
+	refVal, err := refvalToComidTriple(endorsementsStrings)
+	if err != nil {
+		return nil, err
+	}
+
+	evidence, err := evidenceToComidTriple(ec)
+	if err != nil {
+		return nil, err
+	}
+
+	result := handler.CreateAttestationResult(SchemeName)
+
+	appraisal := result.Submods[SchemeName]
+
+	appraisal.TrustVector.InstanceIdentity = ear.NoClaim
```

Comment:
Quick (ignorant) question: Does a verified certificate contain information about the SEV-SNP instance?
If so, this should be populated accordingly (i.e., `ear.TrustworthyInstanceClaim`).